### ISSUE TITLE
url does not include port

### ISSUE THREAD

**Issue by martymoose** [fwestenberg/reolink_dev#70]
installed reolink from the hacs instructions
when i check why its not working i notice the url port number in not there and admin is spelt wrong
do i have to add it some where?

log viewer
2020-09-26 10:44:01 ERROR (stream_worker) [libav.rtmp] Cannot open connection tcp://192.168.1.15:0
2020-09-26 10:44:01 ERROR (stream_worker) [homeassistant.components.stream.worker] Stream connection failed: rtmp://192.168.1.15:None/bcs/channel0_main.bcs?channel=0&stream=0&user=admim&password=xxxxxx
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/stream/worker.py", line 48, in stream_worker
    _stream_worker_internal(hass, stream, quit_event)
  File "/usr/src/homeassistant/homeassistant/components/stream/worker.py", line 65, in _stream_worker_internal
    container = av.open(stream.source, options=stream.options)
  File "av/container/core.pyx", line 354, in av.container.core.open
  File "av/container/core.pyx", line 225, in av.container.core.Container.__cinit__
  File "av/container/core.pyx", line 257, in av.container.core.Container.err_check
  File "av/error.pyx", line 336, in av.error.err_check
av.error.ValueError: [Errno 22] Invalid argument: 'rtmp://192.168.1.15:None/bcs/channel0_main.bcs?channel=0&stream=0&user=admim&password=xxxxxxx'; last error log: [rtmp] Cannot open connection tcp://192.168.1.15:0

and Config
camera:
- platform: reolink_dev
  host: 192.168.1.15
  username: admin
  password: !secret reolink_camera15
  name: frontdoor (optional, default Reolink Camera)
  stream: main or sub (optional, default main)
  protocol: rtmp or rtsp (optional, default rtmp)
  channel: NVR camera channel (optional, default 0)
  scan_interval: 5 (optional, default 30s)

binary_sensor:
  platform: template
  sensors:
    motion_frontdoor:
      friendly_name: Camera frontdoor
      device_class: motion
      entity_id: camera.frontdoor
      value_template: "{{ is_state('camera.frontdoor', 'motion') }}"
      delay_off: 
          seconds: 30
  
cheers marty

**Comment by fwestenberg** [comment]
Hi Marty,

The name ADNIM can only come from your local config somehow. Perhaps you should try reboot first of your HA?
Regarding the port number: this is fetched from your camera config, it should be on the network tab I believe. Please post a screenshot of this page here.

**Comment by DerekMaggio** [comment]
I ran into this error as well with the new Home Assistant Version 0.115.5. 

I am not sure if the cause is the same but I found a workaround to fix it.  

It seems that if your password has an `@` the resulting uri cannot be parsed. If you change your password so that it does not have  `@` it works just fine. 

Also this was not an issue for me in 0.112.2. I see that there have been some changes in the stream library lately. Maybe the culprit is there?
